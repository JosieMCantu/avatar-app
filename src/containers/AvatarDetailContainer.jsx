import React from 'react';
import {useState, useEffect} from 'react';
import {useParams} from 'react-router-dom';
import {fetchSingleCharacter} from '../services/avatarApi';
import CharacterDetail from '../components/presentational/CharacterDetail';

function AvatarDetailContainer() {
    const [character, setCharacter] = useState({});
    const [loading, setLoading] = useState(true);

    const {id} = useParams();

    useEffect(() => {
        fetchSingleCharacter(id)
        .then(setCharacter)
        .finally(() => setLoading(false));
    }, [id]);

    return (
        <>
            <CharacterDetail {...character} />
        </>
    )
}

export default AvatarDetailContainer;