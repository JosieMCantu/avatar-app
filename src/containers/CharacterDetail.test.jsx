import React from 'react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import {render, screen, getByText, waitFor} from '@testing-library/react';
import {userEvent} from '@testing-library/user-event';
import { BrowserRouter as Router, Route, MemoryRouter, } from 'react-router-dom';
import AvatarDetailContainer from './AvatarDetailContainer';


const server = setupServer(
    rest.get('https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68d9', (req, res, ctx) => {
        return res(ctx.json({
            "_id": "5cf5679a915ecad153ab68d9",
            "allies": [
                "Wan"
            ],
            "enemies": [
                "Jaya"
            ],
            "photoUrl": "https://vignette.wikia.nocookie.net/avatar/images/7/76/Aye-aye_spirit.png/revision/latest?cb=20131019224533",
            "name": "Aye-aye spirit",
            "affiliation": "Spirit World"
        }))
    })
)

describe('Avatar Container Test', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());

    it('renders a container to the screen', async () => {
        render(
            <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68d9']}>
                <Route path='/:id'>
                    <AvatarDetailContainer />
                </ Route>
            </MemoryRouter>
        );

        const detail = await screen.findByRole('list', {name: 'character'});
        expect(detail).toMatchSnapshot();
        return waitFor(() => {
            screen.getByAltText('Aye-aye spirit');
            screen.getByText('Spirit World');
          })

    })
})