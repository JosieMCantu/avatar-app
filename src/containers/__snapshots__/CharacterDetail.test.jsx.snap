// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Container Test renders a container to the screen 1`] = `
<ul
  aria-label="character"
>
  <li>
    <h1>
      Detail Page
    </h1>
    <img />
    <p />
    <p />
    <p />
  </li>
</ul>
`;
